app/celery/scheduled_tasks.py

```diff
@@ -221,15 +221,16 @@ def _get_dynamodb_comp_pen_messages(
     https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html#querying-and-scanning
     """
 
+    is_processed_index = 'is-processed-index'
+
     filters = (
-        Attr('is_processed').eq(False)
-        & Attr('payment_id').exists()
+        Attr('payment_id').exists()
         & Attr('payment_id').ne(-1)
         & Attr('paymentAmount').exists()
         & Attr('has_duplicate_mappings').ne(True)
     )
 
-    results = table.scan(FilterExpression=filters, Limit=message_limit)
+    results = table.scan(FilterExpression=filters, Limit=message_limit, IndexName=is_processed_index)
```

Comment:
I am confused on this one because of my knowledge gap on how indexing works. I see in the original query we are filtering for records that have `is_processed` is `False`. 

In the updated query - I see we are indexing on `is_processed_index`. The way I understand the index is that we are now filtering for records that have `is_processed` is `T`. 

Or is `is_processed_index` just filtering for records that have ANY value assigned (is not None)? And at this point, we have not assigned any records as "processed"? 

Thank you.

Reply:
The way the sparse index works is if ANY attribute is in the item then it is in the index, so we are putting 'F' in it but any value would work.  When the item is processed the attribute is removed and then the index no longer has that item.

Reply:
It makes me wonder if the index should be called 'is_not_processed' since it's only supposed to return records with a value but at this point, idk if it's worth the refactor to here and notification-Kafka repo